pyoxidizer/src/py_packaging/distribution.rs

```diff
@@ -360,24 +475,170 @@ impl ParsedPythonDistribution {
+            } else {
+                let dist_prefix = self.base_dir.join("python").join("install");
+
+                copy_dir(&dist_prefix, &venv_base).unwrap();
+
+                let dist_prefix_s = dist_prefix.display().to_string();
+                warn!(
+                    logger,
+                    "copied {} to create hacked base {}", dist_prefix_s, venv_dir_s
+                );
+            };
+        }
+
+        let python_paths = resolve_python_paths(&venv_base, &self.version);
+
+        invoke_python(&python_paths, &logger, &["-m", "ensurepip"]);
+
+        if VENV_MODULE == "virtualenv" {
+            invoke_python(&python_paths, &logger, &["-m", "pip", "install", "virtualenv"]);
+        }
+
+        prepare_hacked_distutils(logger, &python_paths);
+
+        python_paths
+    }
+
+    /// Create a venv from the distribution at path.
+    pub fn create_venv(&self, logger: &slog::Logger, path: &PathBuf) -> PythonPaths {
+        let venv_dir_s = path.display().to_string();
+
+        // This will recreate it, if it was deleted
+        let python_paths = self.create_hacked_base(&logger);
+
+        if path.exists() {
+            warn!(logger, "re-using {} {}", VENV_MODULE, venv_dir_s);
+        } else {
+            warn!(logger, "creating {} {}", VENV_MODULE, venv_dir_s);
+            invoke_python(&python_paths, &logger, &["-m", VENV_MODULE, venv_dir_s.as_str()]);
+        }
+
+        resolve_python_paths(&path, &self.version)
+    }
+
+    /// Create or re-use an existing venv
+    pub fn prepare_venv(
+        &self,
+        logger: &slog::Logger,
+        venv_path: Option<&String>,
+    ) -> (PythonPaths, HashMap<String, String>) {
+        let venv_dir_path = match venv_path {
+            Some(path_str) => PathBuf::from(path_str),
+            None => tempdir::TempDir::new("pyoxidizer-temp-venv")
+                .expect("could not create temp directory")
+                .path()
+                .join("venv"),
+        };
+
+        let python_paths = self.create_venv(logger, &venv_dir_path);
+
+        let mut extra_envs = HashMap::new();
+
+        let prefix_s = python_paths
+            .prefix
+            .canonicalize()
+            .unwrap()
+            .display()
+            .to_string();
+
+        let venv_path_bin_s = python_paths
+            .bin_dir
+            .canonicalize()
+            .unwrap()
+            .display()
+            .to_string();
+
+        let path_separator = if cfg!(windows) { ";" } else { ":" };
+
+        let process_path_s = env::var("PATH").unwrap();
+
+        extra_envs.insert(
+            "PATH".to_string(),
+            format!("{}{}{}", venv_path_bin_s, path_separator, process_path_s),
+        );
+
+        let mut site_packages_s = python_paths
+            .site_packages
+            .canonicalize()
+            .unwrap()
+            .display()
+            .to_string();
+        if site_packages_s.starts_with("\\\\?\\") {
```

Comment:
This prefix is added by `Path.canonicalize`, and I wasnt sure if Python could handle it.
I'm not sure if this is necessary.  Need to try removing it.

Also it seems I need to set `PYTHONHOME`.  I've seen warnings that it wasnt set whilst pip is installing stuff.

Reply:
I read venv `bin/activate` again, and I am not sure about `PYTHONHOME`

Reply:
The message I had seen is appearing on macOS only, and is
```console
...
modifying distutils/unixccompiler.py for oxidation
creating venv /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pyoxidizer-temp-venv.TMkXswfSsJy9/venv
Could not find platform dependent libraries <exec_prefix>
Consider setting $PYTHONHOME to <prefix>[:<exec_prefix>]
Could not find platform dependent libraries <exec_prefix>
Consider setting $PYTHONHOME to <prefix>[:<exec_prefix>]
Collecting gevent
  Downloading ...
```

Reply:
It looks like we are seeing some effects of https://bugs.python.org/issue22213 - when I used `virtualenv` instead of `venv`, the problem goes away.